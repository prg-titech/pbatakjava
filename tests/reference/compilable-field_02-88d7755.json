{
    "basename": "compilable-field_02-88d7755",
    "cmd": "{infile}",
    "infile": "tests/field_02/build.sh",
    "infile_hash": "31b7aaaf0ffa5da13da43752533db3d02652c906e01ca7ea2013753b",
    "stdout": "compilable-field_02-88d7755.stdout",
    "stdout_hash": "d1d6b882c1b90df87daf77ee7b57f78c74e91d121f2e3611584827e3",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}