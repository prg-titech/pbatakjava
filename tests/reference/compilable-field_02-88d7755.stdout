
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);

  public float get__x__float();

  public float set__x(float toAssign);

  public float get__y__float();

  public float set__y(float toAssign);
}

public interface Point_Factory {
  public Point make();
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1() {
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public float get__x__float() {
    return 0;
  }

  public float set__x(float toAssign) {
    return 0;
  }

  public float get__y__float() {
    return 0;
  }

  public float set__y(float toAssign) {
    return 0;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make() {
    return new Point_v1();
  }
}

public class Point_v2 implements Point {
  public float x;

  public float y;

  public Point_v2() {
  }

  public float get__x__float() {
    return this.x;
  }

  public float set__x(float toAssign) {
    return this.x = toAssign;
  }

  public float get__y__float() {
    return this.y;
  }

  public float set__y(float toAssign) {
    return this.y = toAssign;
  }

  public int get__x() {
    return 0;
  }

  public int set__x(int toAssign) {
    return 0;
  }

  public int get__y() {
    return 0;
  }

  public int set__y(int toAssign) {
    return 0;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make() {
    return new Point_v2();
  }
}

public class Test {
  public static void main(String[] args) {
    Point p = new Point_v2();
    float pX = p.get__x__float();
  }
}
